### PR TITLE
Desktop: Update in memory seed usage

### PR DESCRIPTION
**Pull request by rihardsgravis**
# Description

- Add missing manual Garbage Collector trigger after seed usage
- Switch to byte array for all code base

## Type of change

- Bug fix (a non-breaking change which fixes an issue)
- Enhancement (a non-breaking change which adds functionality)

# How Has This Been Tested?

Tested on macOS using `lldb`

# Checklist:

- [x] My code follows the style guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: src/desktop/src/ui/views/account/Tools.js [hunk 1]
```diff
@@ -51,13 +51,23 @@ class Addresses extends PureComponent {
...
+    componentDidUpdate(prevProps) {
+        const { wallet, ui } = this.props;
+
+        if (
+            prevProps.isTransitioning === ui.isTransitioning &&
+            prevProps.isAttachingToTangle === ui.isAttachingToTangle &&
+            prevProps.balanceCheckFlag === wallet.balanceCheckFlag &&
+            prevProps.ui.isSyncing === ui.isSyncing
+        ) {
+            return false;
```
Comment (laumair) [inline src/desktop/src/ui/views/account/Tools.js]: Is there a reason for returning a boolean in `componentDidUpdate`?

Comment (rihardsgravis) [reply src/desktop/src/ui/views/account/Tools.js]: No, will update

File: src/desktop/src/libs/helpers.js [hunk 2]
```diff
@@ -13,6 +13,68 @@ export const capitalize = (input) => {
...
+ */
+const charToByte = (char) => {
+    return '9ABCDEFGHIJKLMNOPQRSTUVWXYZ'.indexOf(char);
+};
+
+const byteToTrit = (byte) => {
+    return trytesTrits[byte % 27];
+};
+
+module.exports = {
```
Comment (rajivshah3) [inline src/desktop/src/libs/helpers.js:74]: Was there an issue with exporting the constant itself?

File: src/desktop/src/ui/views/wallet/Receive.js [hunk 3]
```diff
@@ -128,7 +128,10 @@ class Receive extends React.PureComponent {
                     >
                         <p>
                             {receiveAddress.split('').map((char, index) => {
-                                const scrambleChar = scramble[index] > 0 ? byteToChar(scramble[index]) : null;
+                                const scrambleChar =
+                                    scramble[index] > 0
+                                        ? '9ABCDEFGHIJKLMNOPQRSTUVWXYZ'.charAt(scramble[index] % 27)
```
Comment (rajivshah3) [inline src/desktop/src/ui/views/wallet/Receive.js]: Might be easier if we have this charset stored in another file

**Review by rajivshah3 — APPROVED** [review]

File: src/desktop/src/ui/views/wallet/Receive.js (hunk 3 above)
Comment (rihardsgravis) [reply src/desktop/src/ui/views/wallet/Receive.js]: We actually, do it's `charToByte` function at `libs/helpers`. Will update to use that.

File: src/desktop/src/libs/helpers.js (hunk 2 above)
Comment (rihardsgravis) [reply src/desktop/src/libs/helpers.js:74]: `libs/helpers` is now used also on the main process and that is not going through Babel, so no ES6, here :/

**Pull request merged**